src/twisted/names/test/test_dns.py

```diff
@@ -585,6 +625,38 @@ def test_MX(self):
                 preference=1, name=b'example.com'))
 
 
+    def test_TSIG(self):
+        """
+        The byte stream written by L{dns.Record_TSIG.encode} can be used by
+        L{dns.Record_TSIG.decode} to reconstruct the state of the original
+        L{dns.Record_TSIG} instance.
+        """
+        mac = (b'\x00\x01\x02\x03\x10\x11\x12\x13'
+               b'\x20\x21\x22\x23\x30\x31\x32\x33')
+        rr = dns.Record_TSIG(algorithm='hmac-md5.sig-alg.reg.int',
+                             time=1515548975,
+                             originalID=42, fudge=5,
+                             MAC=mac)
+        self._recordRoundtripTest(rr)
+        rdata = (b'\x08hmac-md5\x07sig-alg\x03reg\x03int\x00'
+                 b'\x00\x00\x5a\x55\x71\x2f\x00\x05\x00\x10' +
+                 mac + b'\x00\x2A\x00\x00\x00\x00')
+        self._recordEncodingTest(rr, rdata)
+
+        rr = dns.Record_TSIG(algorithm='hmac-sha256',
+                             time=4511798055,  # More than 32 bits
```

Comment:
why more than 32bit... I think that this needs a separate test case with a dedicated docstring to explain what we are doing here :)

Reply:
My main reason for the second set of values is to test the case where the optional `error` and `otherData` fields are present. I used a larger time stamp because that field is a 48-bit number, and in my experience it's common for a bug to show up when a field's value crosses a 32-bit boundary.

I suppose it wouldn't hurt anything to split this method into two, but it seems to me that would make the code harder to read, not easier. They're testing essentially the same thing (whether the record's encode/decode methods work).